anomalib/core/model/anomaly_module.py

```diff
@@ -41,21 +41,19 @@ def __init__(self, params: Union[DictConfig, ListConfig]):
         self.save_hyperparameters(params)
         self.loss: torch.Tensor
         self.callbacks: List[Callback]
-        self.register_buffer("threshold", torch.tensor(params.model.threshold.default))  # pylint: disable=not-callable
-        self.threshold: torch.Tensor
+
+        self.image_threshold = AdaptiveThreshold(self.hparams.model.threshold.image_default)
+        self.pixel_threshold = AdaptiveThreshold(self.hparams.model.threshold.pixel_default)
+
+        self.training_stats = AnomalyScoreDistribution()
```

Comment:
Would it be an idea to also rename `self.training_stats` to something like `self.training_distribution`